### PR TITLE
Fix/tao 8009/allow sharedstimulus import with revision

### PR DESCRIPTION
**Pull request by siwane**

### CODE REVIEW

File: controller/MediaManager.php [hunk 1]
```diff
@@ -95,37 +75,43 @@ public function editInstance()
             $this->setData('fileurl', $url);
             $this->setData('mimeType', $mimeType);
 
-        }catch(\tao_models_classes_FileNotFoundException $e){
+        } catch(\tao_models_classes_FileNotFoundException $e) {
```
Comment (martijn-tao) [inline controller/MediaManager.php]: Following the PSR-2 guidelines there should be a space after the `catch` keyword

File: model/FileImporter.php [hunk 2]
```diff
@@ -118,17 +125,16 @@ public function import($class, $form, $userId = null)
...
                     }
                 } else {
-                    $zipImporter = new ZipImporter();
-                    $zipImporter->setServiceLocator($this->getServiceLocator());
-                    $report = $zipImporter->import($class, $form, $userId);
+                    $report = $this->getZipImporter()->import($class, $form, $userId);
                 }
             } else {
                 // editing existing media
                 if (!\helpers_File::isZipMimeType($fileInfo['type'])) {
```
Comment (martijn-tao) [inline model/FileImporter.php]: This `if` statements can be merged into the enclosing `else`

`} elseif (!\helpers_File::isZipMimeType($fileInfo['type'])) {`

File: model/SharedStimulusPackageImporter.php [hunk 3]
```diff
@@ -78,7 +80,7 @@ public function import($class, $form, $userId = null)
      * @return Report
      * @throws \common_exception_NotAcceptable
      */
-    public function edit(Resource $instance, $form)
+    public function edit(Resource $instance, $form, $userId = null)
```
Comment (martijn-tao) [inline model/SharedStimulusPackageImporter.php]: The parameter `$userId` is not described in the method's docblock

File: model/SharedStimulusPackageImporter.php [hunk 4]
```diff
@@ -224,11 +230,13 @@ protected function storeSharedStimulus($class, $lang, $xmlFile, $userId = null)
      * @param \core_kernel_classes_Resource $instance the instance to edit
      * @param string $lang language of the shared stimulus
      * @param string $xmlFile File to store
+     * @param null $userId
```
Comment (martijn-tao) [inline model/SharedStimulusPackageImporter.php]: I believe the type hint for this should be `null|string`

File: test/integration/model/MediaSourceTest.php [hunk 5]
```diff
@@ -72,7 +72,7 @@ public function testAdd()
         $resourceProphecy = $this->prophesize(\core_kernel_classes_Resource::class);
         $resourceProphecy->exists()->willReturn(true);
         $resourceProphecy->getUniquePropertyValue(Argument::any())->willReturn($link, $mime);
-        $resourceProphecy->getPropertyValues(Argument::any())->willReturn(0);
+        $resourceProphecy->getPropertyValues(Argument::any())->willReturn([]);
```
Comment (martijn-tao) [inline test/integration/model/MediaSourceTest.php]: When running this full test I receive the following error:

`Use of undefined constant CONFIG_PATH - assumed 'CONFIG_PATH'`

File: test/integration/model/SharedStimulusImporterTest.php [hunk 6]
```diff
@@ -20,24 +20,26 @@
...
 use Prophecy\Argument;
+use Psr\Log\NullLogger;
 use qtism\data\storage\xml\XmlDocument;
 use qtism\data\storage\xml\XmlStorageException;
 use Zend\ServiceManager\ServiceLocatorInterface;
 
-include_once dirname(__FILE__) . '/../../../../tao/includes/raw_start.php';
-
-class SharedStimulusImporterTest extends TaoPhpUnitTestRunner
+class SharedStimulusImporterTest extends TestCase
```
Comment (martijn-tao) [inline test/integration/model/SharedStimulusImporterTest.php]: When running this test I receive the following errors:

```
UnexpectedValueException : DirectoryIterator::__construct(/Users/username/projects/tao/taoMediaManager/test/integration/sample/fs/): failed to open dir: No such file or directory
 /Users/username/projects/tao/taoMediaManager/test/integration/model/SharedStimulusImporterTest.php:61
```

`Use of undefined constant CONFIG_PATH - assumed 'CONFIG_PATH'`

File: model/FileImporter.php (hunk 2 above)
Comment (siwane) [reply model/FileImporter.php]: No I cannot because it handles 2 behaviours

Comment (martijn-tao) [reply model/FileImporter.php]: But you could right?

Currently we have this:
```
if (case one) {
    // Do stuff for case one
} else {
    if (case two) {
        // Do stuff for case two
    } else {
        // Do whatever case remains
    }
}
```

Which would do the same as:
```
if (case one) {
    // Do stuff for case one
} elseif (case two) {
    // Do stuff for case two
} else {
    // Do whatever case remains
}
```

**Pull request merged**